package/MDAnalysis/core/groups.py

```diff
@@ -1876,6 +1932,65 @@ def __getattr__(self, attr):
         raise AttributeError("{cls} has no attribute {attr}"
                              "".format(cls=self.__class__.__name__, attr=attr))
 
+class UpdatingAtomGroup(AtomGroup):
+    """:class:`AtomGroup` subclass that dynamically updates its selected atoms.
+
+    Accessing any attribute/method of an :class:`UpdatingAtomGroup: instance
+    triggers a check for the last frame the group was updated. If the last
+    frame matches the current trajectory frame, the attribute is returned
+    normally; otherwise the group is updated (the stored selections are
+    re-applied), and only then is the attribute returned.
+
+    Parameters
+    ----------
+    base_group : an :class:`AtomGroup` instance on which *selections* are to be
+        applied.
+    selections : a tuple of :class:`Selection` instances, ready to be applied
+        to *base_group*.
+
+    """
+
+    def __new__(cls, *args, **kwargs):
+        # If we're reinstantiated in a non-updating way, we must
+        # become a regular AtomGroup
+        if len(args) + len(kwargs) > 2:
+            return super(UpdatingAtomGroup, cls).__new__(cls,
+                                                         *args, **kwargs)
+        else:
+            return AtomGroup(*args, **kwargs)
+
+    # ix and u aren't used, but must be part of the signature for class
+    # registration purposes.
+    def __init__(self, ix, u, base_group, selections):
+        self._base_group = base_group
+        self._updating_selections = selections
+        self.update_selection()
+
+    def update_selection(self):
+        bg = self._base_group
+        sels = self._updating_selections
+        ix = sum([sel.apply(bg) for sel in sels])._ix
+        super(UpdatingAtomGroup, self).__init__(ix, bg._u)
+        self._lastupdate = bg._u.trajectory.ts.frame
+
+    def __getattribute__(self,name):
+        if not name in ("_base_group","_updating_selections","_lastupdate"):
+            try:
```

Comment:
Universe will remain constant across updates, so this could probably get set at init time.  With `_last_update` we could set that to an impossible value (-1?) on init so that the equality check always fails first time round.  Then this try/except block doesn't have to exist.